src/com/facebook/buck/swift/SwiftDescriptions.java

```diff
@@ -54,6 +59,18 @@ private SwiftDescriptions() {
       BuildTarget buildTarget) {
     output.srcs = args.srcs.transform(
         input -> filterSwiftSources(sourcePathResolver, args.srcs.get()));
+    output.exportedHeaders = args.exportedHeaders.transform(
+         input -> FluentIterable.from(input.getPaths())
+            .uniqueIndex(new Function<SourcePath, Path>() {
+              @Override
+              public Path apply(SourcePath input) {
+                if (input instanceof PathSourcePath) {
+                  return ((PathSourcePath)input).getRelativePath();
+                }
+                return null;
```

Comment:
In theory, this will never be a non `PathSourcePath`.

Instead of `if` you should use `Preconditions.checkArgument(input instanceof PathSourcePath)` to assert this.  Don't return `null` in any case.